src/back/msl/writer.rs

```diff
@@ -689,10 +689,58 @@ impl<W: Write> Writer<W> {
                 write!(self.out, "{}", name)?;
             }
             crate::Expression::Load { pointer } => {
-                // We don't do any dereferencing with `*` here as pointer arguments to functions
-                // are done by `&` references and not `*` pointers. These do not need to be
-                // dereferenced.
-                self.put_expression(pointer, context, is_scoped)?;
+                // Because packed vectors such as `packed_float3` cannot be directly multipied by
```

Comment:
I suppose this would equally apply to `packed_int3` and friends? So the `wrap_float_scalar_kind` needs to be renamed, and the comments adjusted